spec/mixlib/install/backend/bintray_spec.rb

```diff
@@ -105,4 +106,47 @@
       end
     end
   end
+
+  context "architecture extraction" do
+    let(:channel) { :stable }
+    let(:product_name) { "chef" }
+
+    it "extracts the architecture from file name correctly" do
+      {
+        "x86_64" => [
+          "chef_12.8.1-1_amd64.deb",
+          "chef_12.8.1-1_x64.deb",
+          "chef_12.8.1-1_x86_64.deb",
+          "chef_12.8.1-1.dmg",
+          "chef_12.8.1-1.sh",
+        ],
+        "i386" => [
+          "chef_12.8.1-1_i386.deb",
+          "chef_12.8.1-1_i686.deb",
+          "chef_12.8.1-1_x86.deb",
+          "chef_12.8.1-1_i86pc.deb",
+          "chef_12.8.1-1.msi",
+        ],
+        "powerpc" => [
+          "chef-12.8.1-1.powerpc.bff",
+        ],
+        "sparc" => [
+          "chef-12.8.1-1.sparc.solaris",
+          "chef-12.8.1-1.sun4u.solaris",
+          "chef-12.8.1-1.sun4v.solaris",
```

Comment:
Let's add a few of the legacy `solaris2.5.9` artifacts and a few `solaris2.5.10` in the `i386` section.